### ISSUE TITLE
socket_tests.exe crashed

### ISSUE THREAD

**Issue by rectified95** [microsoft/ebpf-for-windows#2723]
### Describe the bug

Observed a crash in a CI run for an infra PR that doesn't make changes to the product.

### Steps taken to reproduce bug

https://github.com/microsoft/ebpf-for-windows/actions/runs/5721091658/job/15503367288

### Expected behavior

test doesn't crash

### Actual outcome

test crashed

### Additional details
Log file containing many failures, 
[TestLog.txt](https://github.com/microsoft/ebpf-for-windows/files/12225754/TestLog.txt)

Dump+logs: https://microsoft.sharepoint.com/:u:/t/eBPF-vteam/EVbI_5NHZP1HjIOwr2yW1m4Bdc0Vp5xqq3YpqeWwJQucdQ?e=VVvdHH

**Comment by rectified95** [comment]
This is an appverifier failure due to a double closure of the same socket. Credit to Alan:

```
// First close of handle:
void
_client_socket::close()
{
    if (socket != INVALID_SOCKET) {
        closesocket(socket);
    }
}

// Second close of the handle:
_base_socket::~_base_socket()
{
    if (socket != INVALID_SOCKET) {
        closesocket(socket);
    }
}
```
Setting the socket to INVALID_SOCKET fixes it.

**Comment by rectified95** [comment]
Fixed in #2722
